miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/database/MiniAppSecureDatabaseImpl.kt

```diff
@@ -0,0 +1,128 @@
+package com.rakuten.tech.mobile.miniapp.storage.database
+
+import android.annotation.SuppressLint
+import android.content.Context
+import androidx.annotation.NonNull
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.SupportSQLiteOpenHelper
+import com.rakuten.tech.mobile.miniapp.storage.util.MiniAppDatabaseEncryptionUtil
+import net.sqlcipher.database.SupportFactory
+import java.io.IOException
+import java.sql.SQLException
+
+internal enum class MiniAppDatabaseStatus {
+    DEFAULT,
+    INITIATED,
+    OPENED,
+    CLOSED,
+    READY,
+    BUSY,
+    FAILED
+}
+
+/**
+ * Database Implementation Wrapper
+ */
+abstract class MiniAppSecureDatabaseImpl(
```

Comment:
I think, we can define it as `internal abstract class` ?

Reply:
Addressed.